test/integrationTests/hoverProvider.integration.test.ts

```diff
@@ -0,0 +1,57 @@
+/*--------------------------------------------------------------------------------------------- 
+ *  Copyright (c) Microsoft Corporation. All rights reserved. 
+ *  Licensed under the MIT License. See License.txt in the project root for license information. 
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fs from 'async-file';
+import * as vscode from 'vscode';
+import * as path from 'path';
+
+import poll from './poll';
+import { should, expect } from 'chai';
+import testAssetWorkspace from './testAssets/testAssetWorkspace';
+
+const chai = require('chai'); 
+chai.use(require('chai-arrays')); 
+chai.use(require('chai-fs')); 
+
+suite(`Test Hover Behavior ${testAssetWorkspace.description}`, function() {
+    suiteSetup(async function() { 
+        should();
+
+        let csharpExtension = vscode.extensions.getExtension("ms-vscode.csharp"); 
+        if (!csharpExtension.isActive) { 
+            await csharpExtension.activate();
+        }
+        
+        await csharpExtension.exports.initializationFinished;
+    });
+
+
+    test("Hover returns the correct XML", async () => {                
```

Comment:
Can we make this more specific? Perhaps `HoverProvider replaces \n with \n\n`? Hover does not return XML, and this test is validating our newline replacement strategy.